cluster-autoscaler/core/scaledown/legacy/wrapper.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package legacy
+
+import (
+	"time"
+
+	"k8s.io/autoscaler/cluster-autoscaler/processors/nodes"
+	"k8s.io/autoscaler/cluster-autoscaler/processors/status"
+	"k8s.io/autoscaler/cluster-autoscaler/simulator"
+	"k8s.io/autoscaler/cluster-autoscaler/utils/errors"
+
+	apiv1 "k8s.io/api/core/v1"
+	policyv1 "k8s.io/api/policy/v1beta1"
+	klog "k8s.io/klog/v2"
+)
+
+// ScaleDownWrapper wraps legacy scaledown logic to satisfy scaledown.Planner &
+// scaledown.Actuator interfaces.
+type ScaleDownWrapper struct {
```

Comment:
I can see how this works for now, but I think we'll need to refactor TryToScaleDown anyway in order to re-use parts of it. Once we do, I'm not sure how much point there is to having a wrapper like this instead of just making scaleDown object compliant with the interface.
That being said a wrapper is a nice way to allow us to iterate on the implementation and not do entire refactor in a single humongous PR. I'm 100% fine with this PR being based on a wrapper and possibly dropping the wrapper after follow-up PRs make needed refactors.